presto-parquet/src/main/java/com/facebook/presto/parquet/cache/MetadataReader.java

```diff
@@ -373,7 +373,11 @@ private static void readTypeSchema(Types.GroupBuilder<?> builder, Iterator<Schem
                 }
                 typeBuilder = primitiveBuilder;
             }
-            if (element.isSetConverted_type()) {
+
+            if (element.isSetLogicalType()) {
+                getLogicalTypeAnnotationWithParameters(element.getLogicalType()).ifPresent(typeBuilder::as);
+            }
```

Comment:
nit: after this refactor, the `if(element.isSetLogicalType())` block below is redundant, we can remove it.